ote/src/cljs/ote/app/controller/route/route_list.cljs

```diff
@@ -75,11 +141,23 @@
 
   DeleteRouteResponse
   (process-event [{response :response} app]
-    (let [filtered-map (filter #(not= (::transit/id %) (int response)) (get app :routes-vector))]
-      (assoc app :routes-vector filtered-map
-                 :flash-message (tr [:common-texts :delete-route-success])
-                 :routes-changed? true)))
+    (let [routes (get-in app [:routes :routes-vector])
+          filtered-routes (filter
+                            #(not= (::transit/id %) (int response))
+                            routes)]
+      (-> app
+          (assoc-in [:routes :routes-vector] filtered-routes)
+          (assoc :flash-message (tr [:common-texts :delete-route-success]))
+          (assoc :routes-changed? true))))
 
   DeleteRouteResponseFailed
   (process-event [{response :response} app]
     (assoc app :flash-message-error (tr [:common-texts :delete-route-error]))))
+
+(define-event InitRouteList []
+  {}
+  (comm/get! "routes/routes" {:on-success (tuck/send-async! ->LoadRoutesResponse)})
+  app)
+
+(defmethod routes/on-navigate-event :routes [_ app]
```

Comment:
I recall `on-navigate-event` is not triggered on page reload, navigation via browser history or by navigation by address bar (e.g. bookmark).
Should implementation take those cases into account, when `InitRouteList`ist not triggered?

Reply:
In all of those cases on-navigate is called. Tested it in this view.
1. I reloaded page -> on-navigate fired
2. Went to next page, pressed "back" -> on-navigate fired
3. Copied url to new tab -> on-navigate fired.